ModularTegustation/tegu_mobs/lc13_resurgence_clan_mobs.dm

```diff
@@ -0,0 +1,266 @@
+	mob_biotypes = MOB_ROBOTIC
+	gender = NEUTER
+	speech_span = SPAN_ROBOT
+	emote_hear = list("creaks.", "emits the sound of grinding gears.")
+	maxHealth = 500
+	health = 500
+	death_message = "falls to their knees as their lights slowly go out..."
+	melee_damage_lower = 11
+	melee_damage_upper = 15
+	mob_size = MOB_SIZE_HUGE
+	robust_searching = TRUE
+	stat_attack = HARD_CRIT
+	a_intent = INTENT_HARM
+	see_in_dark = 7
+	vision_range = 12
+	aggro_vision_range = 20
+	melee_damage_type = RED_DAMAGE
+	damage_coeff = list(BRUTE = 1, RED_DAMAGE = 0.8, WHITE_DAMAGE = 1.2, BLACK_DAMAGE = 1.5, PALE_DAMAGE = 2)
+	attack_sound = 'sound/weapons/purple_tear/stab2.ogg'
+	butcher_results = list(/obj/item/food/meat/slab/robot = 1, /obj/item/food/meat/slab/sweeper = 1)
+	guaranteed_butcher_results = list(/obj/item/food/meat/slab/robot = 2, /obj/item/food/meat/slab/sweeper = 1)
+	silk_results = list(/obj/item/stack/sheet/silk/azure_simple = 1)
+	var/charge = 0
+	var/max_charge = 10
+	var/clan_charge_cooldown = 2 SECONDS
+	var/last_charge_update = 0
+
+
+/mob/living/simple_animal/hostile/clan/proc/GainCharge()
+	if(stat == DEAD)
+		charge = 0
+		return
+
+	if (charge < max_charge)
+		charge += 1
+		ChargeUpdated()
+
+/mob/living/simple_animal/hostile/clan/proc/ChargeUpdated()
+
+/mob/living/simple_animal/hostile/clan/Life()
+	. = ..()
+
+	if (last_charge_update < world.time - clan_charge_cooldown)
+		last_charge_update = world.time
+		GainCharge()
+
+//Clan Member: Scout
+/mob/living/simple_animal/hostile/clan/scout
+	name = "Scout"
+	desc = "A humanoid looking machine wielding a spear... It appears to have 'Resurgence Clan' etched on their back..."
+	attack_verb_continuous = "stabs"
+	attack_verb_simple = "stab"
+	var/max_speed = 1.5
+	var/normal_speed = 3
+	var/max_attack_speed = 4
+	var/normal_attack_speed = 1
+
+/mob/living/simple_animal/hostile/clan/scout/ChargeUpdated()
+	move_to_delay = normal_speed - (normal_speed - max_speed) * charge / max_charge
+	rapid_melee = normal_attack_speed + (max_attack_speed - normal_attack_speed) * charge / max_charge
+	UpdateSpeed()
+	var/flamelayer = layer + 0.1
+	var/flame
+	if(charge > 7)
+		cut_overlays()
+		flame = "scout_blue"
+	else if(charge > 3)
+		cut_overlays()
+		flame = "scout_red"
+	else
+		cut_overlays()
+		return
+	var/mutable_appearance/colored_overlay = mutable_appearance(icon, flame, flamelayer)
+	add_overlay(colored_overlay)
+
+/mob/living/simple_animal/hostile/clan/scout/AttackingTarget()
+	. = ..()
+	if (charge > 0)
+		charge -= 1
+
+/mob/living/simple_animal/hostile/clan/scout/death(gibbed)
+	. = ..()
+	cut_overlays()
+	charge = 0
+
+//Clan Member: Defender
+/mob/living/simple_animal/hostile/clan/defender
+	name = "Defender"
+	desc = "A humanoid looking machine with two shields... It appears to have 'Resurgence Clan' etched on their back..."
+	icon = 'ModularTegustation/Teguicons/resurgence_48x48.dmi'
+	icon_state = "defender"
+	icon_living = "defender"
+	icon_dead = "defender_dead"
```

Comment:
Defender needs their pixels offset so they are not slightly off tile.
```suggestion
	icon_dead = "defender_dead"
	pixel_x = -8
	base_pixel_x = -8
```

Reply:
Fixed